source/renderer/app/components/profile/bug-report/BugReportDialog.js

```diff
@@ -206,24 +216,29 @@ export default class BugReportDialog extends Component<Props, State> {
     },
   });
 
-  submit = (compressedLog: ?string) => {
+  markIsReadyToSubmit = () => {
+    this.setState({ isReadyToSubmit: true }, this.submit);
+  };
+
+  submit = () => {
+
+    const { compressedLog, isReadyToSubmit } = this.state;
+
+    if (!compressedLog || !isReadyToSubmit) {
+      return false;
+    }
+
+    this.setState({ wasSubmitted: true });
+
     this.form.submit({
       onSuccess: (form) => {
-        const { logFiles } = this.props;
-        const logsExist = get(logFiles, ['files'], []).length > 0;
 
         const { email, subject, problem } = form.values();
         const data = {
           email, subject, problem, compressedLog
         };
 
-        if (this.state.showLogs && logsExist && !compressedLog) {
-          // submit request with commpressed logs files
-          this.props.onCompressLogs(this.props.logFiles);
-        } else {
-          // regular submit
-          this.props.onSubmit(data);
-        }
+        this.props.onSubmit(data);
```

Comment:
@daniloprates we shouldn't send the log files if the user has set the "attach-logs" switch to off state.

Reply:
✔︎